ocs_ci/ocs/resources/mcg.py

```diff
@@ -798,21 +798,26 @@ def _check_mirroring():
                         results.append(True)
                     else:
                         results.append(False)
-
-            return all(results)
-
-        try:
-            for mirroring_is_complete in TimeoutSampler(timeout, 5, _check_mirroring):
-                if mirroring_is_complete:
-                    logger.info("All objects mirrored successfully.")
-                    return True
+            current_percentage = (results.count(True) / len(results)) * 100
+            return current_percentage
+
+        mirror_percentage = _get_mirroring_percentage()
+        logger.info(f"{mirror_percentage}% mirroring is done.")
+        previous_percentage = 0
+        while mirror_percentage < 100:
+            previous_percentage = mirror_percentage
+            for i in range(timeout):
```

Comment:
could we use retry here instead of a while loop?

Reply:
@keesturam  We are resetting the values of timeout if we see some/any progress in sync. 
But with the retries, we will have fixed set of retries. In this case, even if there is sync in progress, it will probably fail after retries